torch/optim/lr_scheduler.py

```diff
@@ -18,6 +18,7 @@
     TypedDict,
     Union,
 )
+from weakref import ref
```

Comment:
is this needed? i think it got removed.

Reply:
`ref` is newly used in line 102.
(I merged main branch to this because github says there are some merge conflict)